tests/ssl-opt.sh

```diff
@@ -8874,6 +8874,40 @@ run_test    "TLS 1.3: minimal feature sets - gnutls" \
             -c "<= parse finished message" \
             -c "HTTP/1.0 200 OK"
 
+requires_openssl_tls1_3
+requires_config_enabled MBEDTLS_SSL_PROTO_TLS1_3
+requires_config_enabled MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE
+requires_config_enabled MBEDTLS_DEBUG_C
+requires_config_enabled MBEDTLS_SSL_CLI_C
+requires_config_enabled MBEDTLS_SSL_ALPN
+requires_config_disabled MBEDTLS_USE_PSA_CRYPTO
+run_test    "TLS 1.3: alpn - openssl" \
+            "$O_NEXT_SRV -msg -tls1_3 -num_tickets 0 -no_resume_ephemeral -no_cache -alpn h2" \
+            "$P_CLI debug_level=3 min_version=tls13 max_version=tls13 alpn=h2" \
+            0 \
+            -c "tls13 client state: MBEDTLS_SSL_HELLO_REQUEST(0)"               \
+            -c "tls13 client state: MBEDTLS_SSL_SERVER_HELLO(2)"                \
+            -c "tls13 client state: MBEDTLS_SSL_ENCRYPTED_EXTENSIONS(19)"       \
+            -c "tls13 client state: MBEDTLS_SSL_CERTIFICATE_REQUEST(5)"         \
+            -c "tls13 client state: MBEDTLS_SSL_SERVER_CERTIFICATE(3)"          \
+            -c "tls13 client state: MBEDTLS_SSL_CERTIFICATE_VERIFY(9)"          \
+            -c "tls13 client state: MBEDTLS_SSL_SERVER_FINISHED(13)"            \
+            -c "tls13 client state: MBEDTLS_SSL_CLIENT_FINISHED(11)"            \
+            -c "tls13 client state: MBEDTLS_SSL_FLUSH_BUFFERS(14)"              \
+            -c "tls13 client state: MBEDTLS_SSL_HANDSHAKE_WRAPUP(15)"           \
```

Comment:
```suggestion
            -c "tls13 client state: MBEDTLS_SSL_HELLO_REQUEST"               \
            -c "tls13 client state: MBEDTLS_SSL_SERVER_HELLO"                \
            -c "tls13 client state: MBEDTLS_SSL_ENCRYPTED_EXTENSIONS"       \
            -c "tls13 client state: MBEDTLS_SSL_CERTIFICATE_REQUEST"         \
            -c "tls13 client state: MBEDTLS_SSL_SERVER_CERTIFICATE"          \
            -c "tls13 client state: MBEDTLS_SSL_CERTIFICATE_VERIFY"          \
            -c "tls13 client state: MBEDTLS_SSL_SERVER_FINISHED"            \
            -c "tls13 client state: MBEDTLS_SSL_CLIENT_FINISHED"            \
            -c "tls13 client state: MBEDTLS_SSL_FLUSH_BUFFERS"              \
            -c "tls13 client state: MBEDTLS_SSL_HANDSHAKE_WRAPUP"           \
```
The value of enum might change.